File: app/src/main/java/org/mifos/mobilewallet/data/rbl/api/ApiInterceptor.java
package org.mifos.mobilewallet.data.rbl.api;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * Created by naman on 23/6/17.
 */

public class ApiInterceptor implements Interceptor {

    public static final String HEADER_CONTENT_TYPE = "content-type";
    public static final String HEADER_ACCEPT = "accept";


    @Override
    public Response intercept(Chain chain) throws IOException {
        Request chainRequest = chain.request();
        Request.Builder builder = chainRequest.newBuilder()
                .header(HEADER_CONTENT_TYPE, "application/json");

        builder.addHeader(HEADER_ACCEPT, "application/json");
        Request request = builder.build();
        return chain.proceed(request);
    }
}

File: core/src/main/java/org/mifos/mobilewallet/core/data/fineract/api/ApiInterceptor.kt
package org.mifos.mobilewallet.core.data.fineract.api

import android.text.TextUtils
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

/**
 * Created by naman on 17/6/17.
 */
class ApiInterceptor(private val authToken: String, private val headerTenant: String) :
    Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val chainRequest = chain.request()
        val builder = chainRequest.newBuilder()
            .header(HEADER_TENANT, headerTenant)
        if (!TextUtils.isEmpty(authToken)) {
            builder.header(HEADER_AUTH, authToken)
        }
        val request = builder.build()
        return chain.proceed(request)
    }

    companion object {
        const val HEADER_TENANT = "Fineract-Platform-TenantId"
        const val HEADER_AUTH = "Authorization"
    }
}